frontend/src/javascript/utils/intervalsUtils.js

```diff
@@ -1,5 +1,50 @@
-export function areIntervalsIntersect(aStart, aEnd, bStart, bEnd) {
-    return aEnd > bStart && aStart < bEnd;
+export function areIntervalsIntersectOrTouching(aStart, aEnd, bStart, bEnd) {
+    return aEnd >= bStart && aStart <= bEnd;
+}
+
+export function applyChangeOnInterval(aStart, aEnd, change) {
+    let result = { start: aStart, end: aEnd };
+
+    if (change.end <= aStart) {
+        result.start += change.size;
+        result.end += change.size;
+        return result;
+    }
+
+    if (change.start >= aEnd) {
+        return result;
+    }
+
+    if (change.text.length > 0 && aStart == change.start && aEnd == change.end) {
+        result.start = aStart;
+        result.end = aStart + change.text.length;
+        return result;
+    }
+
+    if (isSubinterval(change.start, change.end, aStart, aEnd)) {
+        result.start = change.start;
+        result.end = change.start;
+        return result;
+    }
+
+    if (isSubinterval(aStart, aEnd, change.start, change.end)) {
+        result.end += change.size;
+        return result;
+    }
+
+    if (change.start >= aStart && change.start <= aEnd) {
+        result.end = change.start;
+        return result;
+    }
+
+    if (change.end >= aStart && change.end <= aEnd) {
```

Comment:
Jaki jest powód tego, że w ciało tego ifa jest robione inaczej niż ciało ifa powyżej? Imo to są symetryczne sytuacje. Co więcej w takim przypadku jak wkleimy fragment typu "x " w miejscu, które zaczyna się przed zmienną, a kończy wewnątrz zmiennej (czyli de facto ten if) to ta śledzona zmienna przeskakuje na ten "x", co wydaje mi się nie do końca intuicyjne.
![image](https://user-images.githubusercontent.com/57133114/210057821-f500fd41-de4b-4f1e-aa84-5afaf3a4cfb2.png)

![image](https://user-images.githubusercontent.com/57133114/210057753-fcb6cec7-eb2f-45cd-af57-d4145b4d6d6f.png)

Co więcej gdyby nie było tego zabezpieczenia, że z `potentialName` wybieramy tylko pierwszą poprawną nazwę zmiennej, to ten przypadek zakończyłby się w taki sposób:
![image](https://user-images.githubusercontent.com/57133114/210058007-a0c9bee8-cd5e-4f73-9033-3f68c53bab35.png)

Reply:
Ok, juz wiem z czego wynika ta róznica, ale mimo to, błąd występuje. Wydaje mi się, że powinno być tak (ale nie analizowałem szczegółowo):
`result.start = change.end + change.size;`
`result.end += change.size;`